onnx/defs/tensor/defs.cc

```diff
@@ -691,6 +691,117 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
+static const char* IndexPut_ver10_doc = R"DOC(
+Given `data`, `updates` and `indices`, write the values provided by `updates` into `data` at corresponding `indices`.
+The input and output have the same shape.
+
+Example 1:
+  data = [
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 0, 0],
+  ]
+  indices = [
+      [0, 1],
+      [1, 2],
+      [4, 1],
+  ]
+  updates = [
+      [1, 2, 3],
+  ]
+  output = [
+      [0, 1, 0],
+      [0, 0, 2],
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 3, 0]
+  ]
+
+Example 2:
+  data = [
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 0, 0],
+  ]
+  indices = [
+      [0],
+      [1],
+      [4],
+  ]
+  updates = [
+      [1, 2, 3],
+  ]
+  output = [
+      [1, 2, 3],
+      [1, 2, 3],
+      [0, 0, 0],
+      [0, 0, 0],
+      [1, 2, 3]
+  ]
+
+Example 3:
+  data = [
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 0, 0],
+  ]
+  indices = [
+      [0],
+      [1],
+      [4],
+  ]
+  updates = [
+      [1], 
+      [2], 
+      [3],
+  ]
+  output = [
+      [1, 1, 1],
+      [2, 2, 2],
+      [0, 0, 0],
+      [0, 0, 0],
+      [3, 3, 3]
+  ]
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    IndexPut,
```

Comment:
I am not sure if `IndexPut` is a good name. How about `Replace`? It's closer to what it's doing in your examples.

Reply:
That's a good name. I'll change the name for Replace